src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceService.scala

```diff
@@ -441,21 +440,11 @@ class ResourceService(
       // leave a tomb stone if the resource type does not allow reuse
       leaveTombStone = !resourceTypes(resource.resourceTypeName).reuseIds
       _ <- accessPolicyDAO.deleteResource(resource, leaveTombStone, samRequestContext)
+      _ <- cloudSyncPolicies(resource, samRequestContext)
```

Comment:
The resource was deleted in the line above, the database calls in `cloudSyncPolicies` won't find anything. I think you need to get the list of groups that will be changed by having the resource removed before deleting the resource then sync them afterwards.

And something else I think I missed: `PostgresGroupDAO.updateGroupUpdatedDateAndVersion` needs to be called for each group updated.

Reply:
Just to make sure I understand what you're asking, I should do something like:
```
affectedPolicies <- accessPolicyDAO.findAffectedPolicies(resource, samRequestContext) //New method
_ <- accessPolicyDAO.deleteResource(resource, leaveTombStone, samRequestContext)
_ <- cloudSyncPolicies(affectedPolicies, samRequestContext) //Update this method to take in policies instead of finding them
_ <- cloudSyncPolicies(updateGroupUpdatedDateAndVersion, samRequestContext)
```
Is that correct?

Reply:
yes but `updateGroupUpdatedDateAndVersion` should come before `cloudSyncPolicies`. The sync process keeps track of the last synchronized version and won't do anything if that version is the same as the current version. So the order of operations should be change the policy/group in the sam db, bump the version, cloud sync.

And here is yet another idea (sorry if I am jerking you around): instead of your new database queries to bulk delete entries, you can list the policies that need to be updated along with which policy is being removed. For each of those pairs (policy changing, policy removed), call `removeSubjectFromPolicy` (in the same class, `ResourceService`). That takes care of the database calls, syncing, everything. The code would look something like 
```
affectedPolicies <- accessPolicyDAO.findAffectedPolicies(resource, samRequestContext) //New method
_ <- affectedPolicies.traverse { case (policyToUpdate, policyToRemove) => removeSubjectFromPolicy(policyToUpdate, policyToRemove, samRequestContext) }
```
where `findAffectedPolicies` (or maybe there is a better name) returns a `IO[List[(FullyQualifiedPolicyId, FullyQualifiedPolicyId)]]`